src/TicketBai/Mews.Fiscalizations.TicketBai/ModelToDtoConverter.cs

```diff
@@ -0,0 +1,395 @@
+
+        public static Dto.TicketBai Convert(SendInvoiceRequest sendInvoiceRequest)
+        {
+            return new Dto.TicketBai
+            {
+                Cabecera = new Cabecera1
+                {
+                    IDVersionTBAI = TicketBaiVersion
+                },
+                Sujetos = Convert(sendInvoiceRequest.Subject),
+                Factura = Convert(sendInvoiceRequest.Invoice),
+                HuellaTBAI = Convert(sendInvoiceRequest.InvoiceFooter),
+                Signature = null // TODO: Convert(sendInvoiceRequest.Signature)
+            };
+        }
+
+        private static Sujetos Convert(Subject subject)
+        {
+            return new Sujetos
+            {
+                Emisor = Convert(subject.Issuer),
+                Destinatarios = subject.Receivers.Select(r => Convert(r)).ToArray(),
+                VariosDestinatariosSpecified = true,
+                VariosDestinatarios = subject.MultipleReceivers.Match(
+                    t => SiNoType.S,
+                    f => SiNoType.N
+                ),
+                EmitidaPorTercerosODestinatario = subject.IssuerType.Map(t => Convert(t)).ToNullable(),
+                EmitidaPorTercerosODestinatarioSpecified = subject.IssuerType.NonEmpty
+            };
+        }
+
+        private static Emisor1 Convert(Issuer issuer)
+        {
+            return new Emisor1
+            {
+                NIF = issuer.Nif.TaxpayerNumber,
+                ApellidosNombreRazonSocial = issuer.Name.Value
+            };
+        }
+
+        private static EmitidaPorTercerosType Convert(IssuerType type)
+        {
+            return type.Match(
+                IssuerType.IssuedByIssuer, _ => EmitidaPorTercerosType.N,
+                IssuerType.IssuedByThirdParty, _ => EmitidaPorTercerosType.T,
+                IssuerType.IssuedByTransactionRecipient, _ => EmitidaPorTercerosType.D
+            );
+        }
+
+        private static IDDestinatario Convert(Receiver receiver)
+        {
+            var name = receiver.Match(
+                local => local.Name,
+                foreign => foreign.Name
+            );
+            var address = receiver.Match(
+                local => local.Address,
+                foreign => foreign.Address
+            );
+            var postalCode = receiver.Match(
+                local => local.PostalCode,
+                foreign => foreign.PostalCode
+            );
+            return new IDDestinatario
+            {
+                ApellidosNombreRazonSocial = name.Value,
+                CodigoPostal = postalCode.Value,
+                Direccion = address.Value,
+                Item = receiver.Match(
+                    local => (object)local.TaxpayerIdentificationNumber.TaxpayerNumber,
+                    foreign => new IDOtro1
+                    {
+                        CodigoPais = foreign.Country.Map(c => CovertCountryType21(c)).ToNullable(),
```

Comment:
Country is optional for foreign? I wouldn't expect that.

Reply:
![image](https://user-images.githubusercontent.com/51375082/177204345-49f75f87-ddde-4d30-a298-b1476017822d.png)

Reply:
Sure, but that doesn't mean we must have it as optional. I wouldn't accept somebody saying `I am actually not from spain, so I don't need to pass the validations. Also I don't know where I'm from` :smile: